src/integrationTest/java/uk/gov/hmcts/reform/roleassignment/controller/RoleAssignmentIntegrationTest.java

```diff
@@ -48,5 +74,142 @@ public void shoudGetRecordsFromRoleAssignmentTable() throws Exception {
         assertEquals(
             "Role assignment actor Id", "123e4567-e89b-42d3-a456-556642445613", actorId);
     }
-}
 
+    @Test
+    @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD, scripts =
+        {"classpath:sql/insert_role_assignment.sql",
+            "classpath:sql/insert_actor_cache_control.sql"})
+    public void shouldGetRoleAssignmentsBasedOnActorId() throws Exception {
+        assertRoleAssignmentRecordSize();
+        final String url = "/am/role-assignments/actors/123e4567-e89b-42d3-a456-556642445612";
+
+        final MvcResult result = mockMvc.perform(get(url)
+                                                     .contentType(JSON_CONTENT_TYPE)
+                                                     .headers(getHttpHeaders()))
+            .andExpect(status().is(200))
+            .andReturn();
+
+        RoleAssignmentResource response = mapper.readValue(
+            result.getResponse().getContentAsString(),
+            RoleAssignmentResource.class
+        );
+        assertEquals(200, result.getResponse().getStatus());
+        assertEquals(
+            "2ef8ebf3-266e-45d3-a3b8-4ce1e5d93b9f",
+            response.getRoleAssignmentResponse().get(0).getId().toString()
+        );
+        assertEquals(
+            "123e4567-e89b-42d3-a456-556642445612",
+            response.getRoleAssignmentResponse().get(0).getActorId().toString()
+        );
+    }
+
+    @Test
+    @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD, scripts = {"classpath:sql/insert_role_assignment.sql"})
+    public void shouldGetRoleAssignmentsBasedOnRoleTypeAndActorId() throws Exception {
+        assertRoleAssignmentRecordSize();
+        final String url = "/am/role-assignments";
+
+        final MvcResult result = mockMvc.perform(get(url)
+                                                     .contentType(MediaType.APPLICATION_JSON)
+                                                     .headers(getHttpHeaders())
+                                                     .param("roleType", "case")
+                                                     .param("actorId", "123e4567-e89b-42d3-a456-556642445612")
+        )
+            .andExpect(status().is(200))
+            .andReturn();
+        String responseAsString = result.getResponse().getContentAsString();
+
+        List<RoleAssignment> response = Arrays.asList(mapper.readValue(responseAsString, RoleAssignment[].class));
```

Comment:
you can use mapper.readValue(responseAsString, new TypeReference<List<RoleAssignment>>(){}); instead of converting array into ArrayList.Try this way, it will work.

Reply:
Done